DXMainClient/DXGUI/Multiplayer/GameLobby/MultiplayerGameLobby.cs

```diff
@@ -626,62 +626,58 @@ protected void Refresh(bool isHost)
             }
         }
 
-        private void HideMapList()
+        /// <summary>
+        /// Change MapList and other controls visibility.
+        /// </summary>
+        /// <param name="isHost">Visibility</param>
+        private void ChangeMapListAndOtherControlsVisibility(bool isHost)
         {
-            lbChatMessages.Name = "lbChatMessages_Player";
-            tbChatInput.Name = "tbChatInput_Player";
-            MapPreviewBox.Name = "MapPreviewBox";
-            lblMapName.Name = "lblMapName";
-            lblMapAuthor.Name = "lblMapAuthor";
-            lblGameMode.Name = "lblGameMode";
-            lblMapSize.Name = "lblMapSize";
-
-            ReadINIForControl(btnPickRandomMap);
-            ReadINIForControl(lbChatMessages);
-            ReadINIForControl(tbChatInput);
-            ReadINIForControl(lbGameModeMapList);
-            ReadINIForControl(lblMapName);
-            ReadINIForControl(lblMapAuthor);
-            ReadINIForControl(lblGameMode);
-            ReadINIForControl(lblMapSize);
-            ReadINIForControl(btnMapSortAlphabetically);
-
-            ddGameModeMapFilter.Disable();
-            lblGameModeSelect.Disable();
-            lbGameModeMapList.Disable();
-            tbMapSearch.Disable();
-            btnPickRandomMap.Disable();
-            btnMapSortAlphabetically.Disable();
-        }
+            string sourceSuffix; // The name of the target control may have a suffix. e.g. lbChatMessages_Player
+            string targetSuffix; // Change the name of the control to match the new suffix.
+            List<XNAControl> needUpdateControls = new List<XNAControl>(); // Controls that will be updated.
```

Comment:
```suggestion
            List<XNAControl> controlsToUpdate = new();
```
Change the occurences accordingly.